src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/SelectExpandNode.cs

```diff
@@ -84,24 +82,102 @@ public SelectExpandNode(IEdmStructuredType structuredType, ODataSerializerContex
         /// <param name="structuredType">The structural type of the resource that would be written.</param>
         /// <param name="model">The <see cref="IEdmModel"/> that contains the given structural type.</param>
         public SelectExpandNode(SelectExpandClause selectExpandClause, IEdmStructuredType structuredType, IEdmModel model)
-            : this(selectExpandClause, structuredType, model, false)
+            : this()
         {
+            Initialize(selectExpandClause, structuredType, model, false);
         }
 
         /// <summary>
-        /// Creates a new instance of the <see cref="SelectExpandNode"/> class describing the set of structural properties,
-        /// nested properties, navigation properties, and actions to select and expand for the given <paramref name="selectExpandClause"/>.
+        /// Gets the list of EDM navigation properties to be expand referenced in the response.
+        /// keeping this is only for non-breaking changes, This should be replaced by "ReferencedProperties" later.
         /// </summary>
-        /// <param name="selectExpandClause">The parsed $select and $expand query options.</param>
-        /// <param name="structuredType">The structural type of the resource that would be written.</param>
-        /// <param name="model">The <see cref="IEdmModel"/> that contains the given structural type.</param>
-        /// <param name="expandedReference">a boolean value indicating whether it's expanded reference.</param>
-        internal SelectExpandNode(SelectExpandClause selectExpandClause, IEdmStructuredType structuredType, IEdmModel model, bool expandedReference)
-            : this()
+        public ISet<IEdmNavigationProperty> ReferencedNavigationProperties
         {
-            Initialize(selectExpandClause, structuredType, model, false);
+            get
+            {
+                if (ReferencedProperties == null)
+                {
+                    return null;
+                }
+
+                return new HashSet<IEdmNavigationProperty>(ReferencedProperties.Keys);
+            }
+        }
+
+        /// <summary>
+        /// Gets the list of EDM nested properties (complex or collection of complex) to be included in the response.
+        /// keeping this is only for non-breaking changes, This should be replaced by "SelectedComplexes".
+        /// </summary>
```

Comment:
should we mark as obsolete?

Reply:
Yes. We can. However, I don't want to introduce breaking changes. Let's leave the change when we do V8?

---
In reply to: [355519340](https://github.com/OData/WebApi/pull/1941#discussion_r355519340) [](ancestors = 355519340)

Reply:
marked it as obsolete

---
In reply to: [355615343](https://github.com/OData/WebApi/pull/1941#discussion_r355615343) [](ancestors = 355615343,355519340)